includes/admin/marketplace-suggestions/class-wc-marketplace-suggestions.php

```diff
@@ -0,0 +1,167 @@
+	}
+
+	/**
+	 * Suggestion data GET handler.
+	 */
+	public static function get_suggestion_json_data_handler() {
+		$suggestion_data = self::get_suggestions_api_data();
+		wp_send_json_success( $suggestion_data );
+	}
+
+	/**
+	 * Return an array of suggestions the user has dismissed.
+	 */
+	public static function get_dismissed_suggestions() {
+		$dismissed_suggestions = array();
+
+		$dismissed_suggestions_data = get_user_meta( get_current_user_id(), 'wc_marketplace_suggestions_dismissed_suggestions', true );
+		if ( $dismissed_suggestions_data ) {
+			$dismissed_suggestions = $dismissed_suggestions_data;
+			if ( ! is_array( $dismissed_suggestions ) ) {
+				$dismissed_suggestions = array();
+			}
+		}
+
+		return $dismissed_suggestions;
+	}
+
+	/**
+	 * POST handler for adding a dismissed suggestion.
+	 */
+	public static function post_add_dismissed_suggestion_handler() {
+		if ( ! check_ajax_referer( 'add_dismissed_marketplace_suggestion' ) ) {
+			wp_die();
+		}
+
+		$post_data       = wp_unslash( $_POST );
+		$suggestion_slug = sanitize_text_field( $post_data['slug'] );
+		if ( ! $suggestion_slug ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions = self::get_dismissed_suggestions();
+
+		if ( in_array( $suggestion_slug, $dismissed_suggestions, true ) ) {
+			wp_die();
+		}
+
+		$dismissed_suggestions[] = $suggestion_slug;
+		update_user_meta(
+			get_current_user_id(),
+			'wc_marketplace_suggestions_dismissed_suggestions',
+			$dismissed_suggestions
+		);
+
+		wp_die();
+	}
+
+	/**
+	 * Render suggestions containers in products list empty state.
+	 */
+	public static function render_products_list_empty_state() {
+		self::render_suggestions_container( 'products-list-empty-header' );
+		self::render_suggestions_container( 'products-list-empty-body' );
+		self::render_suggestions_container( 'products-list-empty-footer' );
+	}
+
+	/**
+	 * Render a suggestions container element, with the specified context.
+	 *
+	 * @param string $context Suggestion context name (rendered as a css class).
+	 */
+	public static function render_suggestions_container( $context ) {
+		include 'templates/container.php';
+	}
+
+	/**
+	 * Should suggestions be displayed?
+	 *
+	 * User can disabled all suggestions via option.
+	 * We also currently only support English suggestions.
+	 *
+	 * @return bool
+	 */
+	public static function allow_suggestions() {
+		$option = get_option( 'woocommerce_allow_marketplace_suggestions', 'yes' );
+		$allow  = ( 'yes' === $option || 'true' === $option || 1 === $option );
+
+		$locale             = get_locale();
+		$suggestion_locales = array(
+			'en_AU',
+			'en_CA',
+			'en_GB',
+			'en_NZ',
+			'en_US',
+			'en_ZA',
+		);
+		$language_supported = in_array( $locale, $suggestion_locales, true );
+
+		return ( $allow && $language_supported );
+	}
+
+	/**
+	 * Pull suggestion data from remote endpoint & cache in a transient.
+	 *
+	 * @return array of json API data
+	 */
+	private static function get_suggestions_api_data() {
```

Comment:
As mentioned previously, rather have this running in a background process at intervals via the WC_Queue system to fetch the data and store it in the DB.